wgsl/index.bs

```diff
@@ -9385,7 +9417,15 @@ Each input-output location can store a value up to 16 bytes in size.
 The byte size of a type is defined using the *SizeOf* column in [[#alignment-and-size]].
 For example, a four-component vector of floating-point values occupies a single location.
 
-IO locations are specified via the [=attribute/location=] attribute.
+IO locations are specified via the [=attribute/location=] attribute. The IO locations of
+[=fragment=] shader [=user-defined output datum|output=]s can also be specified via the both
+[=attribute/location=] attribute and optional [=attribute/blend_src=] attribute when the feature
+[=extension/dual_source_blending=] is enabled. The [=attribute/blend_src=] attribute can only be
+specified on a [=fragment=] shader [=user-defined output datum|output=] and used to specify up to
+two [=fragment=] shader [=user-defined output datum|output=]s for a given color attachment as the
+inputs to the blend equation. By default the value of the [=attribute/blend_src=] attribute for a
+[=fragment=] shader [=user-defined output datum|output=] is `0` when the [=attribute/blend_src=]
+attribute is not specified in the shader.
```

Comment:
Is there value to having this default? Is there some code generation concern that makes this desirable? It seems like it would be clearer to not have it

Reply:
Here I just refer the statement in [Metal (Page 109)](https://developer.apple.com/metal/Metal-Shading-Language-Specification.pdf) ("If index(i) is not specified in the attribute, the default is an index of 0") and [Vulkan](https://registry.khronos.org/vulkan/specs/1.3-khr-extensions/html/chap15.html#interfaces-fragmentoutput) (variables declared without an Index decoration are considered to have an Index decoration of zero), where the corresponding `[=attribute/blend_src=]` both have a default value `0`. Setting a default value can also make us be able to avoid saying something like `The [=blend_src=] attribute of |output| is omitted or 0` (we can directly say `The [=blend_src=] attribute of |output| is 0`).

Reply:
Maybe I'm misunderstanding how this is used. Here is my understanding, please correct me if it's wrong.

1. Non-dual source blending shader
  * Same as today. No new attributes, just location on outputs.
  * Don't think about `blend_src` at all
2. Dual source blending shader
  * `blend_src(1)` must be specified on output location 0
  * The must only be two outputs at location 0
  * Therefore, omitting `blend_src(0)` is saving a bit of typing, but resulting is less clarity

Is there a case where you could specify `blend_src(0)`, but not `blend_src(1)`? If not then the only other reason I could see for the default would be to aid code generation, but I'm having trouble envisioning that scenario.

Reply:
Thanks for your explanation and now I fully understand your meaning. I think you are right and I agree with you.

Reply:
So if we decide to omit `blend_src(0)`, then I think we can just use `blend_src1` instead of the pattern `blend_src1(num)`. The shader will look like:

```
struct FragOut {
    @location(0) color : vec4<f32>,
    @location(0) @blend_src1 blend : vec4<f32>,
}
```

What do you think?

Reply:
`@blend_src1` seems a lot less consistent with the rest of WGSL. `@blend_src(1)` matches up with the `@location(0)` much nicer. It also means we can have other blend numbers if the need ever arose in the future.

Reply:
I've removed the default value of `blend_src`. PTAL, thanks!